src/app/clusters/joint-fabric-administrator-server/joint-fabric-administrator-server.cpp

```diff
@@ -126,15 +118,69 @@ bool emberAfJointFabricAdministratorClusterAddICACCallback(
     return true;
 }
 
-// TODO
 bool emberAfJointFabricAdministratorClusterOpenJointCommissioningWindowCallback(
     chip::app::CommandHandler * commandObj, const chip::app::ConcreteCommandPath & commandPath,
     const chip::app::Clusters::JointFabricAdministrator::Commands::OpenJointCommissioningWindow::DecodableType & commandData)
 {
+#if CHIP_DEVICE_CONFIG_ENABLE_JOINT_FABRIC
     MATTER_TRACE_SCOPE("OpenJointCommissioningWindow", "JointFabricAdministrator");
+    auto commissioningTimeout = System::Clock::Seconds16(commandData.commissioningTimeout);
+    auto & pakeVerifier       = commandData.PAKEPasscodeVerifier;
+    auto & discriminator      = commandData.discriminator;
+    auto & iterations         = commandData.iterations;
+    auto & salt               = commandData.salt;
+
+    Optional<AdministratorCommissioning::StatusCode> status = Optional<AdministratorCommissioning::StatusCode>::Missing();
+    Status globalStatus                                     = Status::Success;
+    Spake2pVerifier verifier;
 
     ChipLogProgress(Zcl, "Received command to open joint commissioning window");
 
+    FabricIndex fabricIndex       = commandObj->GetAccessingFabricIndex();
+    const FabricInfo * fabricInfo = Server::GetInstance().GetFabricTable().FindFabricWithIndex(fabricIndex);
+    auto & failSafeContext        = Server::GetInstance().GetFailSafeContext();
+    auto & commissionMgr          = Server::GetInstance().GetCommissioningWindowManager();
+
+    VerifyOrExit(fabricInfo != nullptr, status.Emplace(AdministratorCommissioning::StatusCode::kPAKEParameterError));
+    VerifyOrExit(failSafeContext.IsFailSafeFullyDisarmed(), status.Emplace(AdministratorCommissioning::StatusCode::kBusy));
+
+    VerifyOrExit(!commissionMgr.IsCommissioningWindowOpen(), status.Emplace(AdministratorCommissioning::StatusCode::kBusy));
+    VerifyOrExit(iterations >= kSpake2p_Min_PBKDF_Iterations,
+                 status.Emplace(AdministratorCommissioning::StatusCode::kPAKEParameterError));
+    VerifyOrExit(iterations <= kSpake2p_Max_PBKDF_Iterations,
+                 status.Emplace(AdministratorCommissioning::StatusCode::kPAKEParameterError));
+    VerifyOrExit(salt.size() >= kSpake2p_Min_PBKDF_Salt_Length,
+                 status.Emplace(AdministratorCommissioning::StatusCode::kPAKEParameterError));
+    VerifyOrExit(salt.size() <= kSpake2p_Max_PBKDF_Salt_Length,
+                 status.Emplace(AdministratorCommissioning::StatusCode::kPAKEParameterError));
```

Comment:
AdministratorCommissioning::StatusCode::kPAKEParameterError, AdministratorCommissioning::StatusCode::kBusy represent status codes specific to the Administrator Commissioning Cluster. Instead, we should return status codes specific to the Joint Fabric Administrator Cluster, as defined in spec-PR https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/11699/files

Reply:
thx